### ISSUE TITLE
Port forwarding 0.0.0.0 issue

### ISSUE THREAD

**Issue by cihat1** [haugene/docker-transmission-openvpn#820]
Hi,

I am using this container, i installed it following the OMV TechnoDads guide. In that video, he told to add -p 9091:9091 as args. in order to connect to the webClient. It works fine i can connect to client. My question is in container details and network settings 

"Ports": {
                "8888/tcp": null,
                "9091/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "9091"
                    }
                ]
            },

Host ip is 0.0.0.0 , is this normal? Haugene told on my other old post, do not set LOCAL_NETWORK=0.0.0.0 to overcome connection issues. Is this the same mistake?

**Comment by SyRaza** [comment]
To explain a little, 0.0.0.0 means all IPv4 addresses on the machine you're using. If your computer/server is has multiple IP addresses, 0.0.0.0 allows you to reach either or both IP addresses. But this can be an issue with routing VPN sometimes. This might be your issue. Try checking from 0.0.0.0 to whatever your network's IP is.

Do you know what your computer's local/LAN IP address is? Depending on your operating system, there are multiple ways to check. Just Google "how to check IP in windows 10/debian/linux/mac"

Say your IP address is `192.168.1.89`, then you should set the local network variable to `LOCAL_NETWORK=192.168.1.0/24` or `LOCAL_NETWORK=192.168.1.0/16`. Depends on your network.

Likewise, if you IP address is 192.168.3.77 then it should look something like, `LOCAL_NETWORK=192.168.3.0/24` or `LOCAL_NETWORK=192.168.3.0/16`.

Your `HostIp` should not be `0.0.0.0`. It should be whatever your machine's IP address is.

**Comment by haugene** [comment]
**TL;DR:**  No - that is not the same mistake. You're good as long as you don't fuck up the LOCAL_NETWORK variable (which is basically a well placed hole in the VPN tunnel to allow you to communicate locally).

I agree with the explanation from @SyRaza. Just want to say that I'm not sure that HostIp as 0.0.0.0 necessarily is wrong. As you say @SyRaza, if the machine has multiple addresses it would make sense to bind the port on 0.0.0.0 instead of a specific IP.

Anyways @cihat1. The other posts I've written where I warn against LOCAL_NETWORK settings are not related to the 0.0.0.0 in it self but rather that people had /0 at the end. That means that `LOCAL_NETWORK=192.168.1.0/0` is **JUST AS BAD** as `LOCAL_NETWORK=0.0.0.0/0`. (bold and caps lock to warn others scanning for answers: DO NOT COPY AND USE THAT)

But as I said, the problem is with the /0. That basically means that none of the bits in the network specification are fixed. Which again means that all addresses are valid local addresses - and that would result in all traffic being routed through your host instead of through your VPN. For more on this read up on CIDR addresses.

**Comment by cihat1** [comment]
Thanks for replies. I built a home server using omv and docker. I created a new docker network called vpn and used that for the haugenetransmissionvpn container by --network=vpn. I used LOCAL_NETWORK=192.168.0.60/24 which is my servers ip. But i can not connect to transmission webui since container network has vpn running so i put -p 9091:9091 as parameter. That put a route from 0.0.0.0 to my container network. I wondered if thats a leak. Actually i wanted to use your container as a vpn base for my other containers radar, sonarr etc. So i put -p mappings for those containers' ui ports to haugene container to be able to reach them. Then i m curious when i see 0.0.0.0 source routing in iptables, if anyone can reach my whole network after mapping all these ports, that was my first question actually.

My intention is to have a docker network with vpn by using haugene container. And setting other containers with --network container:haugene parameter without opening my whole network to outside. If i do this, i cant reach the uis of all these containers since i dont want to put 0.0.0.0:port:port mappings. And i cant manage to set a proxy to bind these container ports to vpn network. Should i use -p port:port for all those containers i use through haugene container? It is also not feaseable since when i build i a new container i rebuild a new haugene container with the new port binding to be able to reach its UI. Is there a better way? 

Sorry for the long post :) i am trying to learn these stuff.
